sdk/blockchain/blockchain.go

```diff
@@ -424,6 +483,121 @@ func (c *Subnet) InitializeProofOfAuthority(
 	return nil
 }
 
+func startSignatureAggregator(binPath string, configPath string, logFile string) (int, error) {
+	// Function to check if port is in use
+	isPortInUse := func() bool {
+		conn, err := net.Dial("tcp", "localhost:8080")
+		if err == nil {
+			conn.Close()
+			return true
+		}
+		return false
+	}
+
+	// Function to kill existing process
+	killExistingProcess := func() error {
+		// Try pkill first
+		cmd := exec.Command("pkill", "-f", "signature-aggregator")
+		if err := cmd.Run(); err == nil {
+			return nil
+		}
+		return nil
+	}
```

Comment:
do we want to reuse signature aggregator instances, or do we kill and restart each time? @felipemadero